### ISSUE TITLE
the floating container component can end up outside of the screen

### ISSUE THREAD

**Issue by ifrim** [TykTechnologies/tyk-ui#266]
see this PR for more details https://github.com/TykTechnologies/tyk-ui/pull/265

**Comment by ifrim** [comment]
Had a closer look at the issue and the following are my findings.

I think that negative values for the position of the floating container are valid and that the fix that was made in #265 actually causes bugs.

The floating container component displays an element (container) that is "anchored" to the element (target) that was clicked/hovered/something else.  The target element can be scrolled outside of the viewport (check the example below). In this case the container should remain "anchored" to the target and not forced to remain behind inside the viewport.

forced inside the viewport:

https://user-images.githubusercontent.com/5877494/204023869-34a772a8-07e7-4f8e-9e9f-e08db78a4924.mp4

not forced - moves with the target element:

https://user-images.githubusercontent.com/5877494/204024000-2285e411-7db7-44d8-b6e9-7cefa327fb1d.mp4


In conclusion the problem with the tooltip being displayed outside the viewport is not a problem with a fix inside the tyk-ui lib but instead a fix should be made in user land.
1. Solution 1 - force the content of the tooltip to have a maximum width
2. Solution 2 - display the tooltip to the side instead of to the top or bottom

Problem:
![tooltip-content-max-width-before](https://user-images.githubusercontent.com/5877494/204024612-447ca290-392f-42fe-a09b-a7d96ac1454a.png)

Solution 1:
![tooltip-content-max-width](https://user-images.githubusercontent.com/5877494/204024672-9a77b9c6-2067-4ffb-bbe8-24b5254d22ac.png)

Solution 2:
![tooltip-position-right](https://user-images.githubusercontent.com/5877494/204024708-7b745a61-080e-4129-ad76-b3254e09026b.png)

**Comment by Yasintrm** [comment]
maybe tooltip can automatically resize/reposition itself.
get tooltip clientrect `tooltipDomEl.getBoundingClientRect()` this gives 
<pre>
{
    "x": -20,
    "y": 8,
    "width": 500,
    "height": 500,
    "top": 8,
    "right": 480,
    "bottom": 508,
    "left": -20
}
</pre>

so if any side is less then zero, reposition itself.

**Comment by ifrim** [comment]
> maybe tooltip can automatically resize/reposition itself. get tooltip clientrect `tooltipDomEl.getBoundingClientRect()` this gives (relative to viewport)
> 
> {
>     "x": -20,
>     "y": 8,
>     "width": 500,
>     "height": 500,
>     "top": 8,
>     "right": 480,
>     "bottom": 508,
>     "left": -20
> }
> so if any side is less then zero, reposition/resize itself.

@Yasintrm this was the original fix from the PR mentioned above, which I argue is not what we should have. 
If you look at the two videos above the first one is the one that implements this readjusting of negative values. The second video shows the behavior I think is what we would want. For the tooltip and the element that triggered it to "stick" together even if they are scrolled outside of the viewport.

Please note that at the moment the tooltip can be shown on hover but it also has a "show on click" mode which leaves the tooltip opened until you close it manually.

**Comment by lghiur** [comment]
@ifrim I think what @Yasintrm means is that your second proposal should be done automatically 
```
Solution 2 - display the tooltip to the side instead of to the top or bottom
```

I don't want to calculate specifically for every situation I use tooltips a max width or so....

**Comment by ifrim** [comment]
ok, will add support for this

**Comment by lghiur** [comment]
this has been fixed in 3.0.15
